src/mlpack/core/util/program_doc.cpp

```diff
@@ -20,36 +21,70 @@ using namespace mlpack::util;
+ * Construct a ProgramName object.  When constructed, it will register itself
  * with IO.  A fatal error will be thrown if more than one is constructed.
  *
  * @param defaultModule Name of the default module.
- * @param shortDocumentation A short two-sentence description of the program,
+ */
+ProgramName::ProgramName(const std::string& programName)
+{
+  // Register this with IO.
+  IO::GetSingleton().doc.programName = std::move(programName);
+}
+
+/**
+ * Construct a ShortDescription object.  When constructed, it will register
+ * itself with IO.  A fatal error will be thrown if more than one is
+ * constructed.
+ *
+ * @param shortDescription A short two-sentence description of the program,
  *     what it does, and what it is useful for.
- * @param documentation Long string containing documentation on how to use the
- *     program and what it is.  No newline characters are necessary; this is
+ */
+ShortDescription::ShortDescription(const std::string& shortDescription)
+{
+  // Register this with IO.
+  IO::GetSingleton().doc.shortDescription = std::move(shortDescription);
+}
+
+/**
+ * Construct a LongDescription object. When constructed, it will register itself
+ * with IO.  A fatal error will be thrown if more than one is constructed.
+ *
+ * @param longDescription Long string containing documentation on 
+ *     what it is.  No newline characters are necessary; this is
  *     taken care of by IO later.
- * @param seeAlso A set of pairs of strings with useful "see also"
- *     information; each pair is <description, url>.
  */
-ProgramDoc::ProgramDoc(
-    const std::string programName,
-    const std::string shortDocumentation,
-    const std::function<std::string()> documentation,
-    const std::vector<std::pair<std::string, std::string>> seeAlso) :
-    programName(std::move(programName)),
-    shortDocumentation(std::move(shortDocumentation)),
-    documentation(std::move(documentation)),
-    seeAlso(std::move(seeAlso))
+LongDescription::LongDescription(
+    const std::function<std::string()>& longDescription)
 {
   // Register this with IO.
-  IO::RegisterProgramDoc(this);
+  IO::GetSingleton().doc.longDescription = std::move(longDescription);
 }
 
 /**
- * Construct an empty ProgramDoc object.
+ * Construct a Example object.  When constructed, it will register itself
+ * with IO.  A fatal error will be thrown if more than one is constructed.
+ *
+ * @param example Documentation on how to use the program.
  */
-ProgramDoc::ProgramDoc()
+Example::Example(
+    const std::function<std::string()>& example)
 {
-  IO::RegisterProgramDoc(this);
+  // Register this with IO.
+  IO::GetSingleton().doc.example.push_back(std::move(example));
+}
+
+/**
+ * Construct a SeeAlso object.  When constructed, it will register itself
+ * with IO.  A fatal error will be thrown if more than one is constructed.
+ *
+ * @param seeAlso A set of pairs of strings with useful "see also"
```

Comment:
Looks like there is no `seeAlso` parameter.

Reply:
Updated with proper parameters. :)